File: app/src/main/java/com/selfmate/mes/selfmate/ui/repo/ResultsDataSource.java
package com.selfmate.mes.selfmate.ui.repo;

import android.arch.lifecycle.LiveData;

import com.selfmate.mes.selfmate.models.Results;
import com.selfmate.mes.selfmate.models.ResultsDao;

import javax.inject.Inject;

/**
 * Created by vicki_mes on 11/20/2017.
 */

public class ResultsDataSource implements ResultsRepository {
    private ResultsDao resultsDao;

    @Inject
    public ResultsDataSource(ResultsDao resultsDao) {
        this.resultsDao = resultsDao;
    }

    @Override
    public LiveData<Results> getResults() {
        return resultsDao.getResults();
    }

    @Override
    public long insert(Results results) {
        return resultsDao.insertResults(results);
    }

    @Override
    public void updateResults(Results results) {
         resultsDao.updateResults(results);
    }
}

File: app/src/main/java/com/selfmate/mes/selfmate/ui/repo/ResultsDataSource.kt
package com.selfmate.mes.selfmate.ui.repo

import android.arch.lifecycle.LiveData

import com.selfmate.mes.selfmate.models.Results
import com.selfmate.mes.selfmate.models.ResultsDao

import javax.inject.Inject

/**
 * Created by vicki_mes on 11/20/2017.
 */

class ResultsDataSource @Inject
constructor(private val resultsDao: ResultsDao) : ResultsRepository {
    override val liveResults: Results
        get() = resultsDao.getLiveResults()

    override val results: LiveData<Results>
        get() = resultsDao.getResults()

    override fun insert(results: Results): Long {
        return resultsDao.insertResults(results)
    }

    override fun updateResults(results: Results) {
        resultsDao.updateResults(results)
    }
}
